examples/experimental.js

```diff
@@ -0,0 +1,20 @@
+const { Fluvio } = require("../dist");
+
+const options = {
+    host: '127.0.0.1',
+    tls: false
+};
+
+async function main() {
+    const flv = new Fluvio();
+
+    await flv.connect(options);
```

Comment:
I'm wondering if it would make more sense to have a flow like this:

```Typescript
interface FluvioConfig {
  host: string,
  tls: {},
}

class Fluvio {
  // public constructor should be async and fallible
  // So instead of returning a Fluvio, maybe some sort
  // of TS-style representation of Result<Fluvio>
  connect = async (config: FluvioConfig): Fluvio => {
    // Only give back Fluvio on success
  };
  private constructor() {
    // Prevent infallible constructor from being used
  }
}

const config: FluvioConfig = {
  host: "cloud.fluvio.io",
  tls: {},
}

const fluvio = await Fluvio.connect();
```

Reply:
I think it is important to note that node-bindgen has opinions as to what the object structure / class should look like. We can add a TypeScript wrapper around the JS that implements such an interface as the above, which I think will be an easier route forward that trying to implement a `private constructor` in node-bindgen.

Reply:
>   // public constructor should be async and fallible

I think this is considered an anti-pattern for JS (and maybe for other languages as well). Technically, I think it is actually possible, but still considered a bad design.

See https://stackoverflow.com/questions/43431550/async-await-class-constructor